src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/WebConfigurationReaderLevelSwitch.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Internal
+{
+    /// <summary>
+    /// The <see cref="LoggingLevelSwitch"/> implementation that runs callback
+    /// when <see cref="IWebAppLogConfigurationReader.OnConfigurationChanged"/> event fires.
+    /// </summary>
+    public class WebConfigurationReaderLevelSwitch : LoggingLevelSwitch
+    {
+        /// <summary>
+        /// The log level at which the logger is disabled.
+        /// </summary>
+        private static readonly LogEventLevel LogLevelDisabled = LogEventLevel.Fatal + 1;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="WebConfigurationReaderLevelSwitch"/> class.
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="convert"></param>
+        public WebConfigurationReaderLevelSwitch(IWebAppLogConfigurationReader reader, Func<WebAppLogConfiguration, LogLevel> convert )
+        {
+            reader.OnConfigurationChanged += (sender, configuration) =>
```

Comment:
We never replace or recreate loggers so their lifetime is similar to `IWebAppLogConfigurationReader` lifetime so I think handler leak is not an issue here.